### ISSUE TITLE
HibernateSearchTest fails on EE9 because Hibernate is compiled against javax

### ISSUE THREAD

**Issue by benjamin-confino** [OpenLiberty/open-liberty#15817]
I received this error when enabling EE9 on a CDI FAT test that covers some JPA integration. It looks like there might be a reference to javax that escapes the transformer. 

```
Caused by: java.lang.NoClassDefFoundError: javax.persistence.spi.PersistenceProvider
	<java classes>
	at com.ibm.ws.jpa.management.JPAPUnitInfo.createEMFactory(JPAPUnitInfo.java:914)
	at com.ibm.ws.jpa.management.JPAPUnitInfo.initialize(JPAPUnitInfo.java:766)
	at com.ibm.ws.jpa.management.JPAPxmlInfo.extractPersistenceUnits(JPAPxmlInfo.java:182)
	at com.ibm.ws.jpa.management.JPAScopeInfo.processPersistenceUnit(JPAScopeInfo.java:88)
	at com.ibm.ws.jpa.management.JPAApplInfo.addPersistenceUnits(JPAApplInfo.java:119)
	at com.ibm.ws.jpa.container.osgi.internal.JPAComponentImpl.processWebModulePersistenceXml(JPAComponentImpl.java:518)
	at com.ibm.ws.jpa.container.osgi.internal.JPAComponentImpl.applicationStarting(JPAComponentImpl.java:305)
	at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:51)
```

**How to Reproduce**
Remove the SkipForRepeat annotation on com.ibm.ws.cdi12.fat.tests.HibernateSearchTest and run the FAT bucket  `com.ibm.ws.cdi.third.party_fat`.

**Comment by benjamin-confino** [comment]
After talking to the jpa team I have been informed that the issue is that the version of hibernate used in our fat bucket was compiled against javax. 

As there is no hibernate release that uses jakarta I'll keep the test disabled until a version of hibernate built against jakarta is released.

**Comment by tevans78** [comment]
So a newer version of hibernate may be needed ... although that might make the repeat more complicated

**Comment by tevans78** [comment]
https://in.relation.to/2021/01/06/hibernate-validator-700-62-final-released/
Version 7.0 looks like the minimum

**Comment by jhanders34** [comment]
You can also transform the hibernate as part of the test.  That is what @Azquelt did when dealing with tests that used derby / db2 that were compiled for javax.

**Comment by benjamin-confino** [comment]
Resolved by #19974 using the transform method.
